### DISCUSSION TITLE
Virtual input API to activate for 5 seconds then to deactivate automatically - how to do so on AXIS 9188?

### DISCUSSION THREAD

**Discussion by naokibizen** [AxisCommunications/discussions#169] [VAPIX] [answered]
Hello AXIS , for the first time, we will use **AXIS A9188** NETWORK I/O RELAY MODULE.
https://www.axis.com/ja-jp/products/axis-a9188

To control the each relay port on the AXIS 9188,  we are now using **Virtual input API .**
https://www.axis.com/vapix-library/subjects/t10175981/section/t10074527/display?section=t10074527-t10036012

**ACTIVATE**
`//myserver/axis-cgi/virtualinput/activate.cgi?schemaversion=1&port=10`

**DE-ACTIVATE**
`//myserver/axis-cgi/virtualinput/deactivate.cgi?schemaversion=1&port=10`

The ISSUE is that we do not know how to  **deactivate automatically**

Although the AXIS event has auto-reverse mechanism to change activate status to deactivate status on the target relay port (say Relay port 10 ), **however virtual port status is still 'active'. so that next virtualinput activate command fails.**

In sum we see status mismatch between AXIS event mechanism and virtual input  status. 
- Please help us how to solve this issue.

**Comment by mattias-kindborg-at-work** [comment]
Hi @naokibizen. Given that this possibly is specific to a certain product I would like you to open an issue in the [technical support helpdesk](https://www.axis.com/support/helpdesk).

**Comment by onie5** [comment]
@naokibizen you may need to make an ACAP utility to accomplish this.

**Comment by naokibizen** [comment, marked as answer]
Thank you for your advice @mattias-kindborg-at-work , @onie5 !!

This is just another ordinary site to used **AXIS9188 and Elevator Relay On/Off,** not a special site.We thought we can Relay On and Relay off by **some scrip ( VAPIX )** at least we can run a **virtual port deactivate command** every 30 seconds. 

We are very new to ACAP utility. We want to try if it is scrip base application development...

**Reply by onie5** [reply]
@naokibizen sorry to overwhelm you with ACAP. Just went thru the VAPIX library and you may try to look at section Port Management API. I noticed in newer firmwares, activating/deactivating will need to send JSON data. You will also find there a timeout option for state changes.

**Reply by naokibizen** [reply]
The problem  definition here is the status mismatch between virtual port and physical port.
We can **run virtual port activate command** to physical port (say Relay 1 on AXIS9188) to be ON **for 5 seconds ( with de-activate to opposite direction )**,  The physical port works fine as the command does. 

**However the virtual port status stays inactive.**

Thus we have to run a deactivate command to the physical Relay (even if the Rely has already been in inactive ) though the virtual port mechanism.

AXIS may says Virtual port and Physical are different. To us it is in-logical and confusing.....

**Reply by onie5** [reply]
@naokibizen I get now what you mean. The I/O naming is a bit confusing and I was referring you to an output port instead of input.
There is an option in the Action Rules of scheduled event, but I don't think it can go as low as seconds. If it can, you can do a cascading rule, rule 1 for triggering virtual input, rule 2 will use schedules then action as "send notification" to disable virtual input.
The other option that I have used before is thru ACAP.

**Reply by naokibizen** [reply]
Thank you for the advice, @onie5 ! 

- **The problem has been solved** by settings '' Action Rules o'' >   "send notification" ( http command )

In sum

- Virtual Input can activate physical Relay I/O ( Here Virtual Input status has **been left ''active''** )
- However Physical I/O status change can also deactivate  Virtual I/O by "send notification". ( virtual Input can became **''deactivate''** )

Then our problem has resolved.
